src/System.Security.Cryptography.Cng/src/Microsoft/Win32/SafeHandles/NCryptSafeHandles.cs

```diff
@@ -53,42 +55,43 @@ private enum OwnershipState {
             Holder
         }
 
-        private OwnershipState m_ownershipState;
+        private OwnershipState _ownershipState;
 
         /// <summary>
         ///     If the handle is a Duplicate, this points at the safe handle which actually owns the native handle.
         /// </summary>
-        private SafeNCryptHandle m_holder;
+        private SafeNCryptHandle _holder;
 
-        protected SafeNCryptHandle() : base(IntPtr.Zero, true) {
+        protected SafeNCryptHandle() : base(IntPtr.Zero, true)
+        {
             return;
         }
 
         /// <summary>
-        ///     Wrapper for the m_holder field which ensures that we're in a consistent state
+        ///     Wrapper for the _holder field which ensures that we're in a consistent state
         /// </summary>
-        private SafeNCryptHandle Holder {
-            get {
-                Contract.Requires((m_ownershipState == OwnershipState.Duplicate && m_holder != null) ||
-                                  (m_ownershipState != OwnershipState.Duplicate && m_holder == null));
-                Contract.Requires(m_holder == null || m_holder.m_ownershipState == OwnershipState.Holder);
+        private SafeNCryptHandle Holder
+        {
+            get
+            {
+                Debug.Assert((_ownershipState == OwnershipState.Duplicate && _holder != null) ||
+                             (_ownershipState != OwnershipState.Duplicate && _holder == null));
+                Debug.Assert(_holder == null || _holder._ownershipState == OwnershipState.Holder);
 
-                return m_holder;
+                return _holder;
             }
 
-            set {
-                Contract.Ensures(m_holder.m_ownershipState == OwnershipState.Holder);
-                Contract.Ensures(m_ownershipState == OwnershipState.Duplicate);
+            set
+            {
 #if DEBUG
-                Contract.Ensures(IsValidOpenState);
-                Contract.Assert(value.IsValidOpenState);
+                Debug.Assert(value.IsValidOpenState);
 #endif
```

Comment:
No, because IsValidOpenState itself is #if DEBUG only.


From: Jeremy Barton [mailto:notifications@github.com]
Sent: Tuesday, August 4, 2015 2:23 PM
To: dotnet/corefx
Cc: Atsushi Kanamori
Subject: Re: [corefx] Implement System.Security.Cryptography.CngKey (#2615)


In src/System.Security.Cryptography.Cng/src/Microsoft/Win32/SafeHandles/NCryptSafeHandles.cs<https://github.com/dotnet/corefx/pull/2615#discussion_r36243600>:

>  #if DEBUG

> -                Contract.Ensures(IsValidOpenState);

> -                Contract.Assert(value.IsValidOpenState);

> +                Debug.Assert(value.IsValidOpenState);

>  #endif

#if DEBUG

Debug.Assert(...);

#endif

Feels a bit redundant; could the #if be removed?

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/corefx/pull/2615/files#r36243600>.